angular/src/lib/topbar/topbar-mobile.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, HostListener, Input } from '@angular/core';
+
+@Component({
+  selector: 'cui-top-bar-mobile',
+  template: `
+    <cui-icon
+      *ngIf="!isMobileOpen"
+      name="list-menu_20"
+      buttonClassName="cui-top-bar__mobile-menu-button"
+      [attr.aria-pressed]="isMobileOpen"
+      [ariaLabel]="openMenuAriaLabel"
+      (click)="handleOpen()"
+    ></cui-icon>
+    <div
+      [ngClass]="['cui-top-bar__mobile', 'cui-tb-mobile', isMobileOpen && 'open' || '']"
+      (click)="shouldCloseOnClick ? handleClose() : null"
```

Comment:
Check the condition in handleClose method

Reply:
There're several places using the handleClose method. And each of the conditions is different. I don't think moving the condition to handleClose is a proper way.